xcodeproj/internal/xcode_targets.bzl

```diff
@@ -261,45 +261,59 @@ def _to_xcode_target_product(product):
         _additional_files = product.framework_files,
     )
 
-def _merge_xcode_target(*, src, dest):
-    """Creates a new `xcode_target` by merging the values of `src` into `dest`.
+def _merge_xcode_targets(*, srcs, dest):
+    """Creates a new `xcode_target` by merging the values of `srcs` into `dest`.
 
     Args:
-        src: The `xcode_target` to merge into `dest`.
+        srcs: The `xcode_target`s to merge into `dest`.
         dest: The `xcode_target` being merged into.
 
     Returns:
         A value as returned by `xcode_targets.make`.
     """
 
-    build_settings = dict(structs.to_dict(src._build_settings))
+    build_settings = {}
+    for src in srcs:
+        dicts.add(
+            structs.to_dict(src._build_settings),
+            build_settings,
+        )
     build_settings = dicts.add(
         structs.to_dict(dest._build_settings),
         build_settings,
     )
 
+    is_swift = False
+    for src in srcs:
+        if src.is_swift:
+            is_swift = True
+            break
+
     return _make_xcode_target(
         id = dest.id,
         label = dest.label,
         configuration = dest.configuration,
-        compile_target = src,
+        compile_target = srcs[0] if srcs else None,
         package_bin_dir = dest._package_bin_dir,
-        platform = src.platform,
-        product = _merge_xcode_target_product(
-            src = src.product,
+        platform = srcs[0].platform if srcs else None,
+        product = _merge_xcode_targets_product(
+            src_products = [src.product for src in srcs],
             dest = dest.product,
         ),
-        is_swift = src.is_swift,
+        is_swift = is_swift,
         test_host = dest._test_host,
         build_settings = build_settings,
-        conlyopts = src._conlyopts or dest._conlyopts,
-        cxxopts = src._cxxopts or dest._cxxopts,
-        swiftcopts = src._swiftcopts or dest._swiftcopts,
+        conlyopts = flatten([src._conlyopts for src in srcs]) or dest._conlyopts,
```

Comment:
Since we are going to guarantee that there is only one swift and one non-swift, can we instead select the swift copts from the swift one and the non-swift copts from the other target? Then we don't need to flatten.